packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -1556,56 +1554,110 @@ class FocusTraversalGroup extends StatefulWidget {
   /// {@macro flutter.widgets.ProxyWidget.child}
   final Widget child;
 
-  /// Returns the focus policy set by the [FocusTraversalGroup] that most
-  /// tightly encloses the given [BuildContext].
+  /// Returns the [FocusTraversalPolicy] that applies to the nearest ancestor of
+  /// the given [FocusNode].
   ///
-  /// It does not create a rebuild dependency because changing the traversal
-  /// order doesn't change the widget tree, so nothing needs to be rebuilt as a
-  /// result of an order change.
+  /// Will return null if no [FocusTraversalPolicy] ancestor applies to the
+  /// given [FocusNode].
+  ///
+  /// The [FocusTraversalPolicy] is set by introducing a [FocusTraversalGroup]
+  /// into the widget tree, which will associate a policy with the focus tree
+  /// under the nearest ancestor [Focus] widget.
+  ///
+  /// This function differs from [maybeOf] in that it only traverses the focus
+  /// tree to determine the policy in effect. The [maybeOf] function first walks
+  /// up the widget tree, to find the nearest ancestor [Focus] or [FocusScope]
+  /// widget, and then calls this function with the focus node associated with
+  /// that widget to determine the policy in effect.
+  ///
+  /// See also:
+  ///
+  /// * [maybeOf] for a similar function that will return null if no
+  ///   [FocusTraversalGroup] widget ancestor is found in the widget tree.
+  static FocusTraversalPolicy? maybeOfNode(FocusNode node) {
+    return _getGroupNode(node)?.policy;
+  }
+
+  static _FocusTraversalGroupNode? _getGroupNode(FocusNode node) {
+    while (node.parent != null) {
```

Comment:
shouldn't we test for node != null instead?
it's strange to require that the node we return has a parent, although you might argue that a _FocusTraversalGroupNode might never be the root node

Reply:
`node.parent != null` is just another way of saying "while it's not the root node" and the root node is always `FocusManager.instance.rootScope`, which is a FocusScopeNode, and can't be one of these.